src/mbgl/util/mapbox.cpp

```diff
@@ -121,7 +135,8 @@ std::string normalizeTileURL(const std::string& url, const std::string& accessTo
         return url;
     }
 
-    return baseURL + "v4/" + url.substr(sizeof("mapbox://tiles/") - 1) + "?access_token=" + accessToken;
+    auto query = normalizeQuery(url);
+    return baseURL + "v4/" + url.substr(sizeof("mapbox://tiles/") - 1, query.second - sizeof("mapbox://tiles/") + 1) + "?access_token=" + accessToken + normalizeQuery(url).first;
```

Comment:
Hmm, actually it won't throw, that's only if the first argument to `substr` is greater than the size. But can you add a test case anyway?